kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -66,7 +67,8 @@ public SnapshotImpl(
     this.protocol = protocol;
     this.metadata = metadata;
     this.inCommitTimestampOpt = Optional.empty();
-    this.snapshotReport = SnapshotReportImpl.forSuccess(snapshotContext);
+    this.snapshotContext = snapshotContext;
+    this.snapshotReport = null;
```

Comment:
can we do Lazy<SnapshotReport> snapshotReport so that we don't need to initial it to null?